eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/sorted/immutable/ImmutableSortedBagImpl.java

```diff
@@ -792,6 +794,12 @@ public int hashCode()
         return counter.getCount();
     }
 
+    @Override
+    public RichIterable<T> distinctView()
+    {
+        return ImmutableTreeSet.wrap(this.elements, this.comparator);
```

Comment:
Here, you could use `return ArrayAdapter.adapt(this.elements).asUnmodifiable();`

I realize that methods on the view like contains() would perform better if it could use a comparator. But if we need that, we should create a new array-as-set view rather than change an immutable type into a public/mutable one.

Reply:
It does not make it mutable: a user of the returned object cannot mutate it.
It's just that, like with `FastList.newListWith`, it is unsafe to modify the originally passed array to the wrap method, so it only concerns the creator of the object.

If you think anyway that it's best to introduce a new array-as-set class, then I will indeed go with ArrayAdapter and we can maybe create a ticket for improving thing in a later iteration.

Reply:
I just remembered I already had it implemented in my previous tentative to solve this issue, I will just do that then!

Reply:
done, can you check the naming of the new classes (one for unsorted and one for sorted) and their implementations? I just brought back what I had done for #797.

Reply:
Actually, sorry about the back and forth @motlin, but as I was sharing in https://github.com/eclipse/eclipse-collections/pull/1210#discussion_r841204941, I believe it's a bad idea to bring those classes. I will just go with the `ArrayAdapter` in both case and create a ticket to improve the situation later on. IMO the solution is really to use `ImmutableTreeSet`, but there is no rush to decide or find an alternative :)